facebook/mysql-5.6
# Add stall counters to 'SHOW GLOBAL STATUS' output.

@update-submodule: rocksdb

Summary:
The stall counters are already shown in 'SHOW ENGINE ROCKSDB STATUS
output' but this command is a bit expensive to run and you have to parse
the output to monitor these values. This branch adds these counters to
structured output of SHOW GLOBAL STATUS command. 
Also fixed the bug in INFORMATION_SCHEMA.ROCKSDB_COMPACTION_STATS  which showed only integer values but not float.

The rocksdb module was updated to bring [this](https://github.com/facebook/rocksdb/pull/2794#pullrequestreview-59667574) change  which has the following changes to GetMapProperty method:
1. Changed values from double to string
2. Change existing keys to start with "compaction." prefix for all keys returned by old version of GetMapProperty.
3. Added stall values with keys starting with "io_stall." prefix

Test plan:
* Check that compaction level metrics are still published to
INFORMATION_SCHEMA.ROCKSDB_COMPACTION_STATS properly including real double values.
BEFORE:
| __system__ | L0    | Rnp1GB         |     0 |
| __system__ | L0    | Score          |     0 |
| __system__ | L0    | SizeBytes      |  1433 |
AFTER:
| __system__ | L0    | Rnp1GB         |     0 |
| __system__ | L0    | Score          |  0.25 |
| __system__ | L0    | SizeBytes      |  1433 |
* Check the output of the command:
```
mysql> show status like 'rocksdb_stall_%';
+-----------------------------------------------------+-------+
| Variable_name                                       | Value |
+-----------------------------------------------------+-------+
| rocksdb_stall_level0_slowdown                       | 2     |
| rocksdb_stall_level0_slowdown_with_compaction       | 4     |
| rocksdb_stall_level0_numfiles                       | 10    |
| rocksdb_stall_level0_numfiles_with_compaction       | 12    |
| rocksdb_stall_stop_for_pending_compaction_bytes     | 16    |
| rocksdb_stall_slowdown_for_pending_compaction_bytes | 14    |
| rocksdb_stall_memtable_compaction                   | 6     |
| rocksdb_stall_memtable_slowdown                     | 8     |
| rocksdb_stall_total_stop                            | 32    |
| rocksdb_stall_total_slowdown                        | 24    |
| rocksdb_stall_micros                                | 0     |
+-----------------------------------------------------+-------+
11 rows in set (0.00 sec)
```
(In order to have some non-zero numbers, I've initialized internal
RocksDB stall stats with 1-8 numbers. They all doubled in output as
there are two column families in my instance: default and system.
The stall_micros has been introduced previously. total_count doesn't
include level0_slowdown_with_compaction and numfiles_with_compaction
as only their total counterparts are included in this counter.)
